### PR TITLE
Align activity Ethereum chart

### PR DESCRIPTION
**Pull request by michalsidzej**
Resolves L2B-1427

### CODE REVIEW

File: packages/frontend/src/build/createApi.ts [hunk 1]
```diff
@@ -77,13 +77,27 @@ function getActivityChart(
   apiChart: ActivityApiChart,
   ethereumChart: ActivityApiChart,
 ): FrontendActivityChart {
-  const length = Math.min(apiChart.data.length, ethereumChart.data.length)
+  const lastProjectTimestamp = apiChart.data.at(-1)?.[0]
```
Comment (antooni) [inline packages/frontend/src/build/createApi.ts:80]: write this logic as a pure function and add a test for it

File: packages/frontend/src/build/createApi.ts [hunk 2]
```diff
@@ -77,13 +77,27 @@ function getActivityChart(
...
+    (x) => x[0].toNumber() === lastProjectTimestamp.toNumber(),
+  )
+  if (ethChartTimestampIndex === -1) {
+    throw new Error('No matching timestamp in ethereum chart')
+  }
+  const alignedEthChartData = ethereumChart.data.slice(
+    0,
+    ethChartTimestampIndex + 1,
+  )
+  const length = Math.min(apiChart.data.length, alignedEthChartData.length)
```
Comment (antooni) [inline packages/frontend/src/build/createApi.ts:94]: are we sure that both Ethereum and projects data start from the same timestamp?

**Review by antooni — CHANGES_REQUESTED** [review]
Add tests

File: packages/frontend/src/build/createApi.ts (hunk 2 above)
Comment (michalsidzej) [reply packages/frontend/src/build/createApi.ts:94]: Fair point, tested, works :)

File: packages/frontend/src/build/createApi.ts [hunk 3]
```diff
@@ -73,17 +73,31 @@ export function outputCharts(
...
   return {
```
Comment (antooni) [inline packages/frontend/src/build/createApi.ts:95]: this function would be much more readable if it only said: cut the data and return, write everything above return as a seperate function

**Review by antooni — CHANGES_REQUESTED** [review]
Good work, please address some small issues.

**Review by antooni — APPROVED** [review]
LGTM

**Pull request merged**